src/shared/services/fields/fields.utils.ts

```diff
@@ -242,6 +247,82 @@ export class FieldsUtils {
     return !hasNonNullPrimitive ? objectRefs.some(y => this.containsNonEmptyValues(y)) : hasNonNullPrimitive;
   }
 
+  /**
+   * handleNestedDynamicLists()
+   * Reassigns list_item and value data to DynamicList children
+   * down the tree. Server response returns data only in
+   * the `value` object of parent complex type
+   *
+   * EUI-2530 Dynamic Lists for Elements in a Complex Type
+   *
+   * @param jsonBody - { case_fields: [ CaseField, CaseField ] }
+   */
+   public static handleNestedDynamicLists(jsonBody: { case_fields: CaseField[] }): any {
+
+    if (jsonBody.case_fields) {
```

Comment:
Is the if statement necessary? i.e. if there are none it just won't go into the foreach.

Reply:
I copy-pasted this code from CasesService to a more common location. Additionally, I sort of remember `foreach` causing an error when `case_fields` doesn't exist.